### PR TITLE
Adding PushSubscriptionOptions

### PR DESCRIPTION
**Pull request by rachelandrew**
Adds `PushSubscriptionOptions` and also updates the main Push API page and PushSubscription.options which returns this object.

I removed a couple of very outdated bits of browser compat info from the Push API page. If we really need those they ought to be in BCD, but I think they are old enough to not be required.

Reviewer @jpmedley

### CODE REVIEW

File: files/en-us/web/api/pushsubscriptionoptions/applicationserverkey/index.html [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+tags:
+  - API
+  - Property
+  - Reference
+  - applicationServerKey
+  - PushSubscriptionOptions
+---
+<div>{{DefaultAPISidebar("null")}}</div>
+
+<p class="summary">The <strong><code>applicationServerKey</code></strong> null property of the {{domxref("PushSubscriptionOptions")}} interface contains the public key used by the push server.</p>
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/pushsubscriptionoptions/applicationserverkey/index.html:13]: Looks like I have a regression in MDN Helper. Sorry about that.

File: files/en-us/web/api/pushsubscriptionoptions/uservisibleonly/index.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+<div>{{DefaultAPISidebar("null")}}</div>
+
+<p class="summary">The <strong><code>userVisibleOnly</code></strong> read-only property of the {{domxref("PushSubscriptionOptions")}} interface indicates if the returned push subscription will only be used for messages whose effect is made visible to the user.</p>
+
+<h2 id="Syntax">Syntax</h2>
+
+<pre class="syntaxbox">let userVisibleOnly = PushSubscriptionOptions.userVisibleOnly;</pre>
+
+<h3>Value</h3>
+<p>A {{jsxref("Boolean")}} when <code>true</code> indicates that the returned push subscription will only be used for messages whose effect is made visible to the user.</p>
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/pushsubscriptionoptions/uservisibleonly/index.html:20]: This is a judgement call. Regardless, the original was missing a word .

**Review by jpmedley — CHANGES_REQUESTED** [review]
Looking good! Just a few suggestions.

File: files/en-us/web/api/pushsubscriptionoptions/applicationserverkey/index.html [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+title: PushSubscriptionOptions.applicationServerKey
+slug: Web/API/PushSubscriptionOptions/applicationServerKey
+tags:
+  - API
+  - Property
+  - Reference
+  - applicationServerKey
+  - PushSubscriptionOptions
+---
+<div>{{DefaultAPISidebar("null")}}</div>
```
Comment (rachelandrew) [inline files/en-us/web/api/pushsubscriptionoptions/applicationserverkey/index.html:11]:
```suggestion
<div>{{DefaultAPISidebar("Push API")}}</div>
```

File: files/en-us/web/api/pushsubscriptionoptions/uservisibleonly/index.html [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+title: PushSubscriptionOptions.userVisibleOnly
+slug: Web/API/PushSubscriptionOptions/userVisibleOnly
+tags:
+  - API
+  - Property
+  - Reference
+  - userVisibleOnly
+  - PushSubscriptionOptions
+---
+<div>{{DefaultAPISidebar("null")}}</div>
```
Comment (rachelandrew) [inline files/en-us/web/api/pushsubscriptionoptions/uservisibleonly/index.html:11]:
```suggestion
<div>{{DefaultAPISidebar("Push API")}}</div>
```

Comment (rachelandrew) [inline files/en-us/web/api/pushsubscriptionoptions/uservisibleonly/index.html]:
```suggestion
<div>{{DefaultAPISidebar("Push API")}}</div>
```

**Review by jpmedley — APPROVED** [review]

**Pull request merged**